extensions/wikia/PageHeader/ActionButton.class.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use CuratedContentHelper;
+
+class ActionButton {
+	private $contentActions;
+	private $buttonAction;
+	private $pageStatsService;
+	private $title;
+
+	public function __construct( \Title $title ) {
+		$skinVars = \F::app()->getSkinTemplateObj()->data;
+		$this->contentActions = $skinVars['content_actions'];
+		$this->pageStatsService = \PageStatsService::newFromTitle( $title );
+		$this->title = $title;
+
+		$this->prepareActionButton();
+	}
+
+	public function getButtonAction(): array {
+		$this->buttonAction['data-tracking'] = $this->buttonAction['id'];
+		return $this->buttonAction;
+	}
+
+	public function getDropdownActions(): array {
+		// items to be added to "edit" dropdown
+		$actions = [
+			'edit',
+			've-edit',
+			'formedit',
+			'history',
+			'move',
+			'protect',
+			'unprotect',
+			'delete',
+			'undelete',
+			'replace-file',
+			'talk',
+			'edit-mobile-main-page'
+		];
+
+		// Enable to modify actions list on dropdown
+		wfRunHooks( 'PageHeaderDropdownActions', [ &$actions ] );
+
+		$ret = [];
+		foreach ( $actions as $action ) {
+			if ( isset( $this->contentActions[$action] ) ) {
+				$ret[$action] = $this->contentActions[$action];
+				if ( isset( $ret[$action]['id'])) {
+					$ret[$action]['data-tracking'] = $ret[$action]['id'] . '-dropdown';
+				}
+			}
+		}
+
+		return $ret;
+	}
+
+	private function prepareActionButton() {
+		global $wgTitle, $wgUser, $wgRequest, $wgEnableCuratedContentExt;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)